src/App.tsx

```diff
@@ -1,24 +1,211 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import {
+  FC,
+  useEffect,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+import './App.scss';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos,
+  removeTodo,
+  addTodoToServer,
+  changeTodo,
+} from './api/todos';
+import { TodosList } from './components/TodosList/TodosList';
+import { Todo } from './types/Todo';
+import { ErrorInfo } from './components/ErrorInfo/ErrorInfo';
+import {
+  preparedTodos,
+  getcompletedTodoIds,
+  filteredTodosByCompletion,
+} from './utils/todoUtils';
+import { StatusValue } from './types/StatusValue';
+import { TodoAppHeader } from './components/TodoAppHeader/TodoAppHeader';
+import { TodoAppFooter } from './components/TodoAppFooter/TodoAppFooter';
 
-const USER_ID = 0;
+const USER_ID = 10725;
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [todoFilter, setTodoFilter] = useState<StatusValue>(StatusValue.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [visibleError, setVisibleError] = useState('');
+
+  const fetchTodosFromServer = async () => {
+    try {
+      const fetchedTodos = await getTodos(USER_ID);
+
+      setTodos(fetchedTodos as Todo[]);
+    } catch (error) {
+      setVisibleError('Unable to load a todos');
+    }
+  };
+
+  useEffect(() => {
+    fetchTodosFromServer();
+  }, [tempTodo, loadingTodoIds, visibleError]);
+
+  const completedTodoIds = useMemo(() => (
+    getcompletedTodoIds(todos)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => (
+    preparedTodos(todos, todoFilter)
+  ), [todos, todoFilter]);
+
+  const removeTodos = useCallback(async (todoIds: number[]) => {
+    try {
+      setLoadingTodoIds(prevIds => [...prevIds, ...todoIds]);
+
+      await Promise.all(
+        todoIds.map(async id => {
+          await removeTodo(id);
+        }),
+      );
+
+      setTodos((previousTodos) => (
+        previousTodos.filter(todo => !todoIds.includes(todo.id))
+      ));
+    } catch (error) {
+      setVisibleError('Unable to delete a todo');
+    } finally {
+      setLoadingTodoIds([]);
+    }
+  }, []);
+
+  const handleUpdate = useCallback(async (
+    todoIds: number[],
+    newTitle?: string,
+  ) => {
+    try {
+      setLoadingTodoIds((prevIds) => [...prevIds, ...todoIds]);
+
+      await Promise.all(
+        todoIds.map(async id => {
+          const todoToUpdate = todos.find(todo => todo.id === id);
+
+          if (todoToUpdate) {
+            const updatedTodo = await changeTodo(id, {
+              ...todoToUpdate,
+              completed: newTitle
+                ? todoToUpdate.completed
+                : !todoToUpdate.completed,
```

Comment:
For greater readability of the code, it is worth breaking large pieces of code into small pieces of code.
```
const isTodoCompleted = newTitle ? todoToUpdate.completed : !todoToUpdate.completed;
const updatedTodo = await changeTodo(id, {
              ...todoToUpdate,
              completed: isTodoCompleted 
    ....
```